src/utils.js

```diff
@@ -115,6 +115,54 @@ exports.calcCreepCost = function(body) {
     return result;
 };
 
+function calcEnergyAvailable(roomObjects, energyStructures){
+    return _.sum(energyStructures, id => {
+        if (roomObjects[id] && !roomObjects[id].off && (roomObjects[id].type === 'spawn' || roomObjects[id].type === 'extension')) {
+            return roomObjects[id].energy;
+        } else {
+            return 0;
+        }
+    });
+}
+
+function data(roomObjects, id) {
+    if(!roomObjects[id]) {
+        throw new Error("Could not find an object with ID "+id);
+    }
+    return roomObjects[id];
+}
+
+exports.canCreateCreep = function canCreateCreep(roomObjects, spawn, spawnArgs){
```

Comment:
What's the motivation to keep it in `utils` if it's used only from one single place?

Reply:
Good question. You could probably identify it if you could see the history of the work. I tried a few different approaches to implementing this before settling on the current solution.

As far as I recall, I was originally changing the canCreateCreep function. When we decided to implement the dryRun behavior in order to keep the existing API identical, I extracted this to util. I'm up for moving it somewhere else if you wish.

Reply:
I don't think it should be in its own function now, just put this code as regular checks in `createCreep` method.

Reply:
Done.